x-pack/plugins/lens/public/indexpattern_plugin/rename_columns.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ExpressionFunction } from 'src/legacy/core_plugins/interpreter/types';
+import { KibanaDatatable } from '../types';
+
+interface RemapArgs {
+  idMap: string;
+}
+
+export const renameColumns: ExpressionFunction<
+  'lens_rename_columns',
+  KibanaDatatable,
+  RemapArgs,
+  KibanaDatatable
+> = {
+  name: 'lens_rename_columns',
+  type: 'kibana_datatable',
+  help: 'A helper to rename the columns of a datatable',
+  args: {
+    idMap: {
+      types: ['string'],
+      help:
+        'A JSON encoded object in which keys are the old column ids and values are the corresponding new ones. All other columns ids are kept.',
+    },
+  },
+  context: {
+    types: ['kibana_datatable'],
+  },
+  fn(data: KibanaDatatable, { idMap: encodedIdMap }: RemapArgs) {
+    const idMap = JSON.parse(encodedIdMap) as Record<string, string>;
+    return {
+      type: 'kibana_datatable',
+      rows: data.rows.map(row => {
+        const mappedRow: Record<string, unknown> = {};
+        Object.entries(idMap).forEach(([fromId, toId]) => {
+          mappedRow[toId] = row[fromId];
+        });
+
+        Object.keys(row)
+          .filter(id => !(id in idMap))
+          .forEach(unchangedId => {
+            mappedRow[unchangedId] = row[unchangedId];
+          });
```

Comment:
You're doing two loops here, but I think this could all be done in one loop like this:

```typescript
Object.entries(row).forEach(([id, value]) => {
  if (id in idMap) {
    mappedRow[idMap[id]] = value;
  } else {
    mappedRow[id] = value;
  }
});
```

Reply:
I somehow default to chain filter and map calls but this is definitely easier to understand, thanks.